### PR TITLE
trak: init at 0.0.5

### PR DESCRIPTION
**Pull request by buurro**
## Description of changes

New package: https://github.com/lcfd/trak

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/tr/trak/package.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  src =
+    let
+      repo = fetchFromGitHub {
+        owner = "lcfd";
+        repo = "trak";
+        rev = "v${version}";
+        hash = "sha256-Lc5/i/OuAwHoPo8SeBZe+JSstZov8K/ku7EMyb2I9Ng=";
+      };
+    in
+    "${repo}/cli";
```
Comment (AndersonTorres) [inline pkgs/by-name/tr/trak/package.nix:20]: why not to use `sourceRoot` or a mere `cd cli` instead?

File: pkgs/by-name/tr/trak/package.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  propagatedBuildInputs = with python3Packages; [
+    typer
+  ] ++ typer.optional-dependencies.all;
+
+  build-system = [ python3Packages.poetry-core ];
+
+  meta = with lib; {
+    description = "Keep a record of the time you dedicate to your projects.";
```
Comment (AndersonTorres) [inline pkgs/by-name/tr/trak/package.nix:29]:
```suggestion
    description = "Keep a record of the time you dedicate to your projects";
```

File: pkgs/by-name/tr/trak/package.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    in
+    "${repo}/cli";
+
+  propagatedBuildInputs = with python3Packages; [
+    typer
+  ] ++ typer.optional-dependencies.all;
+
+  build-system = [ python3Packages.poetry-core ];
+
+  meta = with lib; {
```
Comment (AndersonTorres) [inline pkgs/by-name/tr/trak/package.nix:28]:
```suggestion
  meta = {
```

File: pkgs/by-name/tr/trak/package.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    description = "Keep a record of the time you dedicate to your projects.";
+    homepage = "https://github.com/lcfd/trak";
+    license = licenses.agpl3Only;
+    maintainers = with maintainers; [ buurro ];
```
Comment (AndersonTorres) [inline pkgs/by-name/tr/trak/package.nix:32]:
```suggestion
    license = lib.licenses.agpl3Only;
    maintainers = with lib.maintainers; [ buurro ];
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by buurro** [comment]
Thanks for the feedback!

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -2999,6 +2999,11 @@
     githubId = 37375448;
     name = "Buildit";
   };
+  buurro = {
+    github = "buurro";
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:3003]: email needed

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Result of `nixpkgs-review pr 304910` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>trak</li>
    <li>trak.dist</li>
  </ul>
</details>

**Review by Aleksanaa — APPROVED** [review]
approve

**Pull request merged**